frappe/erpnext
# feat: adds item filter in buying doctypes on selected supplier

Changes made:
- Add 'Filter by Supplier' checkbox to each of the Buying doctypes (Purchase Order, Purchase Receipt, Purchase Invoice, Supplier Quotation)
- Filter checkbox is only displayed if a supplier is selected
- Add supplier items query to ERPNext `queries.py` to get all items where that supplier is listed either as default supplier or in supplier details child table
- Add filter logic to `buying.js` so that it applies to all buying doctypes

Function
- Allow a user to select only items from a specific supplier
- Filter only applied if a supplier is selected and checkbox is active
- Filter is triggered on the checkbox and hence does not affect buying doctypes that do not have the field

Steps to Test
- Ensure one or more items are active and have a supplier listed either as default or in the supplier details child table
- Open any of the buying doctypes as listed above and create a new one
- Set the supplier
- Items should still show all purchasable items in the table dropdown or in select multiple popup box
- Select the checkbox
- Items should filter only to those with that supplier

Screenshots

- Items with supplier set

![image](https://user-images.githubusercontent.com/33743873/100453764-8c9ff480-30e1-11eb-8e1a-af5ef9e40378.png)


- Purchase order with supplier set but checkbox not set

![image](https://user-images.githubusercontent.com/33743873/100454174-6169d500-30e2-11eb-90d4-7ab107a07cf0.png)

![image](https://user-images.githubusercontent.com/33743873/100454207-76deff00-30e2-11eb-9fa4-4396272ee37f.png)


- Purchase order with supplier and checkbox set

![image](https://user-images.githubusercontent.com/33743873/100454242-852d1b00-30e2-11eb-809d-0bce4430b2d1.png)

![image](https://user-images.githubusercontent.com/33743873/100454261-8eb68300-30e2-11eb-97fa-44865e2fb9c7.png)
